app/services/payout/bitflyer_service.rb

```diff
@@ -1,58 +1,69 @@
 module Payout
-  class BitflyerService < Service
-    def perform
-      # Writes a message if we should skip a publisher or not.
-      # Implemented in Payout::Service
-      return if skip_publisher?
+  class BitflyerService
+    def self.build
+      new(refresher: Bitflyer::Refresher.build, payout_utils_class: Payout::Service)
+    end
+
+    def initialize(refresher:, payout_utils_class:)
+      @refresher = refresher
+      @payout_utils_class = payout_utils_class
+    end
+
+    # Change to call as soon as we refactor the other services
+    def perform(publisher:, payout_report:)
+      payout_utils = @payout_utils_class.new(payout_report: payout_report,
+                                             publisher: publisher,)
+
+      return if payout_utils.skip_publisher?
 
       potential_payments = []
 
-      connection = @publisher.bitflyer_connection
+      connection = publisher.bitflyer_connection
       # Sync the connection
-      # connection.sync_connection!
+      @refresher.call(bitflyer_connection: connection)
 
       potential_payments << PotentialPayment.new(
-        payout_report_id: @payout_report&.id,
```

Comment:
@nvonpentz This is the section you are most likely interested in as it details what you'll end up seeing.